src/common/components/heading-element-for-level.tsx

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { NamedFC } from 'common/react/named-fc';
+import * as React from 'react';
+
+export type HeadingLevel = 1 | 2 | 3 | 4 | 5 | 6;
```

Comment:
This vs aliasing `number` is an interesting choice. I was surprised when I saw this that it wasn't causing compilation failures everywhere the PR uses `childHeadingLevel={parentHeadingLevel + 1}`; it turns out, the only reason there aren't errors about that is because `tab-stops-instance-section-props-factory.tsx` and `result-section-content.tsx` both have `headingLevel: number` props still instead of `headingLevel: HeadingLevel`.

I think to enable the `headingLevel + 1` cases properly you need a runtime check somewhere. You could either do:

* Have `HeadingLevel` alias `number` instead of being an enum type, and make `HeadingElementForLevel` throw an error if it's given something that isn't an integer 1-6, or:
* Keep `HeadingLevel` and `HeadingElementForLevel` as is, but replace all the `headingLevel + 1` invocations with invocations of a new function along the lines of `nextHeadingLevel` that encapsulates the addition alongside a "throw if called with 6" check

I lean towards the second one personally but I think either would be ok.

Reply:
if you take the second approach, you might also consider having `HeadingLevel` be non-numeric (eg, `'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'`) to make it harder for consumers to misuse it with "+ and cast"

Reply:
This makes sense. I considered the nextHeadingLevel helper idea originally but didn't go with it because it could easily be ignored. If `HeadingLevel` is made non-numeric, that would make the helper function more useful/harder to skip.

For this PR, I've implemented the first part of the suggested change (performing headingLevel + 1 only in a helper function). I think it would make the most sense to make the non-numeric change to the type in the follow up refactoring PR, since that will eliminate are usage of number-based heading levels entirely.

Reply:
This makes sense. I considered the nextHeadingLevel helper idea originally but didn't go with it because it could easily be ignored. If HeadingLevel is made non-numeric, that would make the helper function more useful/harder to skip.

For this PR, I've implemented the first part of the suggested change (performing headingLevel + 1 only in a helper function). I think it would make the most sense to make the non-numeric change to the type in the follow up refactoring PR, since that will eliminate our usage of number-based heading levels entirely.